### PR TITLE
Regen and save historical states as default 

### LINKED ISSUE #5537: Regen and save historical state is broken
Doesn't get passed archival index 61

```
[2020-04-20 07:53:29]  INFO Saved historical state archivedStateSlot=15872 currentArchivedIndex/totalArchivedIndices=61/71
15872
[2020-04-20 07:53:30] ERROR main: could not regenerate historical states, please retry: could not regenerate historical state transition: could not process block: could not process block header: parent root 0xaa791859a1cefacce1a6449cf0217e316ec63adf8983922d0645acaffcb26219 does not match the latest block header signing root in state 0x6a442f6d5471b9c4493f8afe5dda654596f250d011572e2fa9baa9ef82147d5a
```

### PR DESCRIPTION
**Pull request by terencechain**
This fixes #5537

Change list:
* Removed interaction menu. Set regenerating and saving of historical states as default once `--disable-new-state-mgmt`  flag is off
* Fixed issue #5537 by correctly feed in parent state and not last state

Tested it ~10 times using the DBs in Minio. Can confirm this implementation is reliably working
Screen Shot 2020-04-20 at 11 46 50 AM

### CODE REVIEW

File: beacon-chain/state/stategen/migrate.go [hunk 1]
```diff
@@ -82,7 +82,9 @@ func (s *State) MigrateToCold(ctx context.Context, finalizedSlot uint64, finaliz
 			// could cause issue switching back.
 			if s.beaconDB.HasState(ctx, r) && r != finalizedRoot {
 				if err := s.beaconDB.DeleteState(ctx, r); err != nil {
-					return err
+					// For whatever reason if node is unable to delete a state due to
+					// state is finalized, it is more reasonable to continue than to exit.
+					continue
```
Comment (prestonvanloon) [inline beacon-chain/state/stategen/migrate.go:87]: Want to log a warning?

Comment (terencechain) [reply beacon-chain/state/stategen/migrate.go:87]: sure

File: beacon-chain/db/kv/regen_historical_states.go [hunk 2]
```diff
@@ -54,6 +56,10 @@ func (kv *Store) regenHistoricalStates(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	cacheState, err := lru.New(int(params.BeaconConfig().SlotsPerEpoch) * 2)
```
Comment (rauljordan) [inline beacon-chain/db/kv/regen_historical_states.go:59]: can this cache size be at the top of the file as a const

Comment (prestonvanloon) [reply beacon-chain/db/kv/regen_historical_states.go:59]: It's never used again so we should let this be GC'd after this method returns

**Pull request merged**